includes/framework/QDbBackedSessionHandler.class.php

```diff
@@ -123,6 +123,15 @@ public static function SessionRead($id) {
 			if (!$result_row) // either the data was empty or the row was not found
 				return '';
 			$strData = $result_row['data'];
+
+			if(strstr($objDatabase->Adapter, 'PostgreSql')) {
```

Comment:
Re: per database type handling, the problem is that this is hand-written SQL so it will not really help us here. We do not want to read the database schema every time we do a database operation. QQuery is supposed to help us with this, so perhaps we need to turn this into a QQuery operation.